__tests__/integration.js

```diff
@@ -0,0 +1,391 @@
+const awsServerlessExpress = require('../index')
+const apiGatewayEvent = require('../example/api-gateway-event.json')
+const app = require('../example/app')
+
+const server = awsServerlessExpress.createServer(app)
+const lambdaFunction = {
+  handler: (event, context) => awsServerlessExpress.proxy(server, event, context)
+}
+
+function clone(json) {
+  return JSON.parse(JSON.stringify(json))
+}
+
+function makeEvent(eventOverrides) {
+  const baseEvent = clone(apiGatewayEvent)
+  const headers = Object.assign({}, baseEvent.headers, eventOverrides.headers)
+  delete eventOverrides.headers
+  const root = Object.assign({}, baseEvent, eventOverrides)
+  root.headers = headers
+  return root
+}
+
+function expectedRootResponse() {
+  return makeResponse({
+    "headers": {
+      "content-length": "3747",
+      "content-type": "text/html; charset=utf-8",
+      "etag": "W/\"ea3-WawLnWdlaCO/ODv9DBVcX0ZTchw\""
+    }
+  })
+}
+
+function makeResponse(response) {
+  const baseResponse = {
+    "body": "",
+    "isBase64Encoded": false,
+    "statusCode": 200
+  }
+  const baseHeaders = {
+    "access-control-allow-origin": "*",
+    "connection": "close",
+    "content-type": "application/json; charset=utf-8",
+    "x-powered-by": "Express"
+  }
+  const headers = Object.assign({}, baseHeaders, response.headers)
+  const finalResponse = Object.assign({}, baseResponse, response)
+  finalResponse.headers = headers
+  return finalResponse
+}
+
+describe('integration tests', () => {
+  test('GET HTML (initial request)', (done) => {
+    const succeed = response => {
+      delete response.headers.date
+      expect(response.body.startsWith('<!DOCTYPE html>')).toBe(true)
+      const expectedResponse = expectedRootResponse()
+      delete response.body
+      delete expectedResponse.body
+      expect(response).toEqual(expectedResponse)
+      done()
+    }
+    lambdaFunction.handler(makeEvent({ path: '/', httpMethod: 'GET' }), {
+      succeed
+    })
+  })
+
+  test('GET HTML (subsequent request)', (done) => {
```

Comment:
aws-serverless-express has different code paths for first request (cold start) and subsequent requests. Could clean this up into a function which both of these root tests could call.